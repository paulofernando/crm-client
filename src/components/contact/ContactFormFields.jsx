import React from "react";
import { Form, ToggleButtonGroup, ToggleButton } from "react-bootstrap";

import "../../App.css";

class ContactFormFields extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      alert: "",
      type: "",
      caseOptions: ""
    };
  }

  render() {
    const {
      values,
      errors,
      touched,
      handleChange,
      handleBlur,
      setFieldValue
    } = this.props;

    return (
      <>
        <Form.Group controlId="formFirstName">
          <Form.Label>First Name</Form.Label>
          <Form.Control
            type="text"
            name="firstName"
            placeholder="John"
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.firstName}
            className={touched.name && errors.name ? "error" : null}
          />
          {touched.firstName && errors.firstName ? (
            <div className="error-message">{errors.firstName}</div>
          ) : null}
        </Form.Group>

        <Form.Group controlId="formLastName">
          <Form.Label>Last Name</Form.Label>
          <Form.Control
            type="text"
            name="lastName"
            placeholder="Smith"
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.lastName}
            className={touched.lastName && errors.lastName ? "error" : null}
          />
          {touched.lastName && errors.lastName ? (
            <div className="error-message">{errors.lastName}</div>
          ) : null}
        </Form.Group>

        <Form.Group controlId="formEmail">
          <Form.Label>Email</Form.Label>
          <Form.Control
            type="text"
            name="email"
            placeholder="john@email.com"
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.email}
            className={touched.email && errors.email ? "error" : null}
          />
          {touched.email && errors.email ? (
            <div className="error-message">{errors.email}</div>
          ) : null}
        </Form.Group>

        <Form.Group controlId="formCaseRole">
          <Form.Label>Court Role</Form.Label>
          <br></br>
          <ToggleButtonGroup type="radio" name="caseRole" size="sm">
            <ToggleButton
              variant="outline-primary"
              value={"Accused"}
              onChange={() => setFieldValue("caseRole", "Accused")}
            >
              Accused
            </ToggleButton>
            <ToggleButton
              variant="outline-primary"
              value={"Barrister"}
              onChange={() => setFieldValue("caseRole", "Barrister")}
            >
              Barrister
            </ToggleButton>
            <ToggleButton
              variant="outline-primary"
              value={"Judge"}
              onChange={() => setFieldValue("caseRole", "Judge")}
            >
              Judge
            </ToggleButton>
            <ToggleButton
              variant="outline-primary"
              value={"Prosecutor"}
              onChange={() => setFieldValue("caseRole", "Prosecutor")}
            >
              Prosecutor
            </ToggleButton>
          </ToggleButtonGroup>
        </Form.Group>
      </>
    );
  }
}

export default ContactFormFields;
