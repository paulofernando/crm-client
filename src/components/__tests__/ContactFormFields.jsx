import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ContactFormFields from '../contact/ContactFormFields';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

// -------------------------------

const contact = {
    firstName: "John",
    lastName: "Smith",
    caseRole: "Judge",
    email: "js@js.com"
}

const ContactFormFieldsComponent = (
    <ContactFormFields
        values={contact}
        errors={errors}
        touched={touched} />
)

// -------------------------------

describe('ContactFormFields', () => {

    const errors = {
        title: ""
    }

    const touched = {
        title: ""
    }

    it('renders correctly', () => {
        const wrapper = renderer
            .create(ContactFormFieldsComponent)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

});