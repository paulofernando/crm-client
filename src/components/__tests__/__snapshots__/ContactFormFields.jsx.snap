// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactFormFields renders correctly 1`] = `
Array [
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formFirstName"
    >
      First Name *
    </label>
    <input
      className="form-control"
      id="formFirstName"
      name="firstName"
      placeholder="John"
      type="text"
      value="John"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formLastName"
    >
      Last Name *
    </label>
    <input
      className="form-control"
      id="formLastName"
      name="lastName"
      placeholder="Smith"
      type="text"
      value="Smith"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formEmail"
    >
      Email *
    </label>
    <input
      className="form-control"
      id="formEmail"
      name="email"
      placeholder="john@email.com"
      type="text"
      value="js@js.com"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formCaseRole"
    >
      Court Role *
    </label>
    <br />
    <div
      className="btn-group btn-group-sm btn-group-toggle"
      role="group"
    >
      <label
        className="btn btn-outline-primary"
        disabled={false}
      >
        <input
          autoComplete="off"
          checked={false}
          disabled={false}
          name="caseRole"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="Accused"
        />
        Accused
      </label>
      <label
        className="btn btn-outline-primary"
        disabled={false}
      >
        <input
          autoComplete="off"
          checked={false}
          disabled={false}
          name="caseRole"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="Barrister"
        />
        Barrister
      </label>
      <label
        className="btn btn-outline-primary"
        disabled={false}
      >
        <input
          autoComplete="off"
          checked={false}
          disabled={false}
          name="caseRole"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="Judge"
        />
        Judge
      </label>
      <label
        className="btn btn-outline-primary"
        disabled={false}
      >
        <input
          autoComplete="off"
          checked={false}
          disabled={false}
          name="caseRole"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="Prosecutor"
        />
        Prosecutor
      </label>
    </div>
  </div>,
]
`;
